firebase-firestore/src/main/java/com/google/firebase/firestore/DocumentSnapshot.java

```diff
@@ -178,11 +180,31 @@ public <T> T toObject(
     checkNotNull(
         serverTimestampBehavior, "Provided serverTimestampBehavior value must not be null.");
     Map<String, Object> data = getData(serverTimestampBehavior);
-    return data == null
-        ? null
+    // Ktx decoder goes here!
+    MapEncoder ktxMapEncoder = getKtxMapEncoderById();
+    if (data == null) return null;
+    return isKtxSerializable(valueType)
+        ? ktxMapEncoder.decode(data, valueType, getReference())
```

Comment:
This is my plan for handling Kotlin decoder.

Reply:
What is the purpose of finding a specific encoder by id? can't you use `supports()` instead? i.e.

```java
for(MapEncoder encoder : availableEncoders) {
  if(encoder.supports(value.class)) {
    return encoder.decode(value);
  }
}
// fallback to default
return CustomClassMapper(convertToCustomClass(data, valueType, getReference());
```

The idea is that the first encoder that claims to support the type is used.

Reply:
Thank you, I have discussed with Rodrigo, and we are using `supports()` now.